### DISCUSSION TITLE
Info Panel

### DISCUSSION THREAD

**Discussion by zaymonoid** [BetterThanTomorrow/calva#1071] [Ideas]
The F# VsCode tooling (https://ionide.io/index.html) has this concept called an `info panel`. I haven't seen it anywhere else and dearly miss it (even when using other IDE's for F#). 

I find it exceedingly useful in development and I feel like it could really pair well with Clojure's REPL driven development (especially as a productivity booster for beginners).

The basic idea is have the option to spawn a panel that shows the documentation for any function under the editor caret.

The functionality it offers is:
- Panel can be locked as a reference
- Namespaces referenced in the panel can be explored
- Doc strings, parameter lists, return types are shown

**Screenshot**:
![image](https://user-images.githubusercontent.com/12402727/111571886-41763280-87f3-11eb-9e51-1dd5ef6e895d.png)

**Comment by PEZ** [comment]
Looks awesome! Please file it as an issue. No idea when we will have the time to try get inspired from it, but it's good to have it there, because maybe someone (maybe even you 😄 ) find it and have the time and the will.

I've been wanting to add ClojureDocs support for a while as well, and this might be a good vehicle for that.

**Reply by jwhitlark** [reply]
Is this still under consideration, or has existing functionality mostly caught up to it?  The pop-up docs are great, but I could see using this from time to time when longer consideration is needed.  I wonder if it's something that could be done in joyride?

**Reply by PEZ** [reply]
Something like it can be done with Joyride. There's an example in the Joyride repo for how to open Clojuredocs in the internal browser. As well as for creating web views. Not sure where this falls.

We probably also should consider it for built-in to Calva.
